### DISCUSSION TITLE
Data loss in Elasticsearch

### DISCUSSION THREAD

**Discussion by ben-sec** [Security-Onion-Solutions/securityonion#5595] [Unsupported Versions] [answered]
Hello!

SO 2.3 ran smoothly without any interruptions since the very beginning (June 1, 2021), but now I wanted to take a look at my logs in the past months and discovered this:

![image](https://user-images.githubusercontent.com/81571847/134124870-1bae6313-87b8-416e-826e-f276ab86d8c0.png)

Any ideas what might have happened to my log data in the very beginning (June 1.-18.) and between July 9 - August 8? What logs should I start with to search for the root cause of this data loss? Disk space is not the issue. Do SO Updates delete ES data? Thanks in advance!

Cheers, Ben

**Comment by dougburks** [comment, marked as answer]
Depending on your architecture, you may want to look at logs for filebeat, logstash, redis, and elasticsearch:
https://docs.securityonion.net/en/2.3/filebeat.html#diagnostic-logging
https://docs.securityonion.net/en/2.3/logstash.html#log
https://docs.securityonion.net/en/2.3/redis.html#diagnostic-logging
https://docs.securityonion.net/en/2.3/elasticsearch.html#diagnostic-logging

**Comment by ben-sec** [comment]
I'm working with SO on a daily basis and I would instantly notice if logs won't come in because of logstash, redis or filebeat errors. Is there a way to find evidences in the elasticsearch diagnostic logs that somebody has deleted ES content manually? Are there possible/known scenarios where ES is loosing old data? All ideas are welcome!

Cheers, Ben

**Reply by TOoSmOotH** [reply]
sudo so-elasticsearch-query _cat/indices 

Run that and see if those indices are there but in a closed state.

**Comment by ben-sec** [comment]
On my search node I get a list of 1023 indices of which 805 have the status "close". Not good? How do I open them? Who closed them?

**Reply by dougburks** [reply]
https://docs.securityonion.net/en/2.3/curator.html#configuration

**Reply by ben-sec** [reply]
If curator is closing the indices after 30 days, how does that explain my screenshot above? Why do I still see some data back in June? Will I run into trouble/challenges if I set cur_close_days to 200 (as long as I provide enough disk space)? Thanks a lot for your support, I really appreciate that!

**Reply by dougburks** [reply]
If you have sufficient disk space, then you can certainly try tuning `cur_close_days` and `log_size_limit`.

**Reply by ben-sec** [reply]
Hello Doug,
I set cur_close_days to 200 in the minion config of my manager node, but ES is still closing Indices:

![image](https://user-images.githubusercontent.com/81571847/137317367-86cd9243-da49-43df-abcb-dc2dd2f03c57.png)
(please compare it to the screenshot at the top)

The log_size_limit is set to 8582 GB in the config file of the search node and the /nsm directory is at 4.7 TB of available 10.5 TB. Any ideas what's going on here?

Cheers, Ben

**Reply by TOoSmOotH** [reply]
What version are you running and what is your setup?

**Reply by ben-sec** [reply]
I have recently updatet to SO 2.3.80 and run a distributed environment with a manager, search and forward node. ES is not running in a cluster.

**Reply by dougburks** [reply]
> I set cur_close_days to 200 in the minion config of my manager node

In which file in /opt/so/saltstack/local/pillar/minions/ did you set this value?

**Reply by ben-sec** [reply]
In "my-manager_manager.sls"

**Reply by dougburks** [reply]
Have you tried setting it in the sls file for your search node?

**Reply by ben-sec** [reply]
No. Ah, I just saw that curator is running on the search node, not the manager. In what section do I have to put the cur_close_days setting in the search node SLS file?

**Reply by dougburks** [reply]
Looks like we had some outdated documentation at https://docs.securityonion.net/en/2.3/curator.html#configuration. I've updated it as follows. Please try that and see if it works for you.

Curator defaults to closing indices older than 30 days. To modify this, edit /opt/so/saltstack/local/pillar/global.sls and change the `close` setting for each index under the `elasticsearch:index_settings` section.

**Reply by ben-sec** [reply]
OK, I will give it a try. Thanks for the support!

**Comment by ben-sec** [comment]
Looks like the new settings for the indicies are working as expected and so far the indicies stay open.
